provider/server/exec.go

```diff
@@ -124,16 +123,21 @@ func (e executorExec) Exec(ctx context.Context, in io.Reader, out io.WriteCloser
 		return fmt.Errorf("machine %s has not yet been synced", machineID)
 	}
 
-	uri, err := e.Libvirt.ConnectGetUri()
+	domainXMLData, err := e.Libvirt.DomainGetXMLDesc(domain, 0)
```

Comment:
don't forget implement some protection for multiple connection at the same time to the same machine. It isn't supported use case because we have only one session.